sdk/java/pulumi/src/test/java/com/pulumi/core/internal/UrnTest.java

```diff
@@ -127,7 +136,7 @@ private static Stream<Arguments> testCreateUrnString() {
                         "stack", "project",
                         Optional.of("urn:pulumi:stack::project::parentPackage:parentType$package:type::name"),
                         "package2:type2", "name2",
-                        "urn:pulumi:stack::project::package:type$package2:type2::name2"
+                        "urn:pulumi:stack::project::parentPackage:parentType$package:type$package2:type2::name2"
```

Comment:
I believe the original test case was incorrect because there should be 2 parents in the type since the parent resource itself has a parent.